### PR TITLE
Replace macro with filter for grid

### PR DESCRIPTION
**Pull request by davidpmccormick**
## What is this PR trying to achieve?
Replacing the grid macro with a nunjucks filter.

### CODE REVIEW

File: server/filters/grid.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    if (key.includes('Shift')) {
+      const shiftIndex = key.indexOf('Shift');
+      size = `shift-${key.substr(0, shiftIndex)}`;
+    }
+
+    retString += ` grid__cell--${size}${colsAtSizes[key]}`;
+  }
+
+  return retString;
+}
```
Comment (jamesgorrie) [inline server/filters/grid.js]:
```JS
const sizes = {m:6, xl:10, shiftM:1, shiftXl:1};
  const base = 'grid__cell';
  const t = Object.keys(sizes).map(key => {
    const size = sizes[key];
    const modifier = key.replace(/([a-z])([A-Z])/g, '$1-$2').toLowerCase();
    return `${base}--${modifier}${size}`;
  });


  return [base].concat(t).join(' ');
```

File: server/filters/grid.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export default function grid(colsAtSizes, omitGridCell) {
```
Comment (jamesgorrie) [inline server/filters/grid.js]: I wonder if it's worth calling this `gridClasses` or something?

File: server/filters/grid-classes.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export default function gridClasses(sizes = {}) {
+  const base = 'grid__cell';
+  const t = Object.keys(sizes).map(key => {
```
Comment (jamesgorrie) [inline server/filters/grid-classes.js]: Might be worth calling this something meaningful?

File: server/filters/grid-classes.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export default function gridClasses(sizes = {}) {
```
Comment (jamesgorrie) [inline server/filters/grid-classes.js]: What's the point of setting the default? People should be required to send a set of sizes no?

File: server/views/components/new-site-notice/index.njk [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <div class="row row--yellow row--text-bar new-site-notice">
   <div class="container">
     <div class="grid">
-      <div class="grid__cell grid__cell--shift-l1">
+      <div class="{{ {lShift: 1} | gridClasses }}">
```
Comment (jamesgorrie) [inline server/views/components/new-site-notice/index.njk]: These have now changed to shiftL...

Comment (davidpmccormick) [inline server/views/components/new-site-notice/index.njk]: Good point, thanks

**Pull request merged**